similarity-search/src/main/java/org/phenotips/similarity/internal/SolrSimilarPatientsFinder.java

```diff
@@ -151,16 +155,28 @@ public int compare(PatientSimilarityView o1, PatientSimilarityView o2)
      * @param referencePatient the reference patient
      * @return a query populated with terms from the patient phenotype
      */
+    @SuppressWarnings("unchecked")
     private SolrQuery generateQuery(Patient referencePatient, boolean prototypes)
     {
+        SolrQuery queryt = new SolrQuery();
+        StringBuilder qr = new StringBuilder();
+        if (referencePatient.getDocument() != null) {
+            qr.append("(document:" + ClientUtils.escapeQueryChars(referencePatient.getDocument().toString()) + " )");
+        }
+        queryt.add(CommonParams.Q, qr.toString());
+        SolrDocument doc = search(queryt).get(0);
+        List<String> ancestorsList =  (List<String>) doc.getFieldValue("phenotype_ancestors");
```

Comment:
This fails when the patient is an in-memory (e.g. remote) patient. You cannot rely on the index to get the phenotype_ancestors.